cumulusci/tasks/push/tests/test_push_api.py

```diff
@@ -1,12 +1,541 @@
+import datetime
+import json
 from unittest import mock
+
 import pytest
+from simple_salesforce import SalesforceMalformedRequest
+
+from cumulusci.tasks.push.push_api import (
+    BasePushApiObject,
+    MetadataPackage,
+    MetadataPackageVersion,
+    PackagePushError,
+    PackagePushJob,
+    PackagePushRequest,
+    PackageSubscriber,
+    SalesforcePushApi,
+    batch_list,
+    memoize,
+)
+
+NAME = "Chewbacca"
+SF_ID = "033xxxxxxxxx"
+PUSH_API = "push_api"
+NAMESPACE = "namespace"
+ORG_KEY = "bar"
+
+
+@pytest.fixture
+def sf_push_api():
+    return SalesforcePushApi(mock.Mock(), mock.Mock())  # sf  # logger
+
+
+@pytest.fixture
+def metadata_package():
+    return MetadataPackage(
+        push_api=mock.MagicMock(), name=NAME, sf_id=SF_ID, namespace=NAMESPACE
+    )
+
+
+@pytest.fixture
+def metadata_package_package_push_job():
+    return PackagePushJob(
+        push_api=mock.MagicMock(),
+        request=mock.Mock(),
+        org="00D63000000ApoXEAS",
+        status="Succeeded",
+        sf_id=SF_ID,
+    )
+
+
+@pytest.fixture
+def metadata_package_version(metadata_package):
+    return MetadataPackageVersion(
+        push_api=mock.MagicMock(),
+        package=metadata_package,
+        name=NAME,
+        sf_id=SF_ID,
+        state="Beta",
+        major="1",
+        minor="2",
+        patch="3",
+        build="4",
+    )
+
+
+@pytest.fixture
+def package_push_job():
+    return PackagePushJob(
+        push_api=mock.MagicMock(),
+        request="",
+        org="00DS0000003TJJ6MAO",
+        status="Succeeded",
+        sf_id=SF_ID,
+    )
+
+
+@pytest.fixture
+def package_subscriber():
+    return PackageSubscriber(
+        push_api=mock.MagicMock(),
+        version="1.2.3",
+        status="Succeeded",
+        org_name="foo",
+        org_key="bar",
+        org_status="Succeeded",
+        org_type="Sandbox",
+        sf_id=SF_ID,
+    )
+
+
+@pytest.fixture
+def package_push_error():
+    return PackagePushError(
+        push_api="foo",
+        sf_id=SF_ID,
+        job="Foo",
+        severity="high",
+        error_type="bar",
+        title="foo_bar",
+        message="The foo hit the bar",
+        details="foo bar, foo, foo bar",
+    )
+
+
+@pytest.fixture
+def package_push_request():
+    return PackagePushRequest(
+        push_api=mock.MagicMock(),
+        version="1.2.3",
+        start_time="12:03",
+        status="Succeeded",
+        sf_id=SF_ID,
+    )
+
+
+def test_base_push_format_where():
+    base_obj = BasePushApiObject()
+    field_name = "id_field"
+    sf_id = "006000000XXX000"
+    where_clause = "id=001000000XXX000"
+    base_obj.sf_id = sf_id
+
+    returned = base_obj.format_where(field_name, where_clause)
+    assert "{} = '{}' AND ({})".format(field_name, sf_id, where_clause)
```

Comment:
You forgot `== returned`, so this is merely asserting that there is a non-empty string.

Reply:
Are you sure about this isn't this a soql query? Not python comparison?

Reply:
I know the equal sign here is part of the soql query. I'm saying that essentially this line is

    assert "some string"

which is of course true. I think you want:

    assert "{} = '{}' AND ({})".format(field_name, sf_id, where_clause) == returned

Reply:
Ah ok